hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -138,6 +142,20 @@ protected void refreshTimeline(HoodieTimeline visibleActiveTimeline) {
     this.visibleCommitsAndCompactionTimeline = visibleActiveTimeline.getWriteTimeline();
   }
 
+  /**
+   * Refresh the completion time query view.
+   */
+  protected void refreshCompletionTimeQueryView() {
+    this.completionTimeQueryView = new CompletionTimeQueryView(metaClient);
```

Comment:
this is going to run full reload including archived (slim) timeline. Does it need to happen more than once in the lifecyle of a query or batch write? I see that it can be called by HoodieMetadataFileSystemView as well. We need to be a bit careful here.

Reply:
Yeah, I've been thinking through it for a while and the loading is just as light-weight as the timeline, there should not much cost here, I don't think it deserves to make the loading incremental because that would increase the complexity, fired a JIRA to trace it: https://issues.apache.org/jira/browse/HUDI-6921